stdlib/3.5/pathlib.pyi

```diff
@@ -1,33 +1,32 @@
-# Stubs for pathlib (Python 3.4)
+# Stubs for pathlib (Python 3.5)
 
-from typing import Any, Generator, IO, Optional, Sequence, Tuple, Union
+from typing import Any, AnyStr, Generator, IO, Optional, Sequence, Tuple, Union
 import os
 
-class PurePath:
+class PurePath(object):
     parts = ...  # type: Tuple[str, ...]
-    drive = ... # type: str
-    root = ... # type: str
+    drive = ...  # type: str
+    root = ...  # type: str
     anchor = ...  # type: str
     parents = ...  # type: Sequence[PurePath]
     parent = ...  # type: PurePath
     name = ...  # type: str
     suffix = ...  # type: str
     suffixes = ...  # type: List[str]
     stem = ...  # type: str
-    def __init__(self, *pathsegments: str) -> None: ...
-    def __hash__(self) -> int: ...
+    def __new__(cls, *args: Union[str, PurePath]) -> PurePath: ...
```

Comment:
The actual [source code](https://github.com/python/cpython/blob/0a267c396861944fd14bb46a380a962020375162/Lib/pathlib.py#L614) itself uses `__new__` instead of `__init__`, so I switched to using it. My understanding is that as far as MyPy is concerned the two are effectively equivalent, so I'm happy to switch it back if you'd rather :)